helpers/checkpoint_logger/__init__.py

```diff
@@ -432,15 +436,22 @@ def submit_subflow(self: _Self, metric: str, start: T, end: T) -> _Self:
         if duration:
             sentry_sdk.set_measurement(metric, duration, "milliseconds")
             duration_in_seconds = duration / 1000
+            repo_id = self.context and self.context.repo_id
             PROMETHEUS_HANDLER.log_subflow(
-                flow=self.cls.__name__, subflow=metric, duration=duration_in_seconds
+                flow=self.cls.__name__,
+                subflow=metric,
+                duration=duration_in_seconds,
+                repo_id=repo_id,
             )
 
         return self
 
 
 def from_kwargs(
-    cls: type[T], kwargs: MutableMapping[str, Any], strict: bool = False
+    cls: type[T],
+    kwargs: MutableMapping[str, Any],
+    strict: bool = False,
+    context: CheckpointContext = None,
```

Comment:
for this particular case, you could just pick `repo_id` (aka `repoid`?) out of the `kwargs`.

Reply:
I don't think that would work off the bat as the `repoid` has been specified as a keyword parameter, thus not belonging to **kwargs when it's supplied to the checkpoints functionality, for instance here https://github.com/codecov/worker/blob/5514b97f08b80b0615ea4122e1c3edab89dd3d45/tasks/upload.py#L300-L301. Unless you meant re-adding it to the kwargs object before we supply it. As an add-on, I created the `context` to a) type the supplied params and b) serve as an object that has items additional to the checkpoint flow - I'm open to a different approach, I chose this one for readability + separation of concerns. (Although I will rename repo_id to repoid to be consistent with our model definition)

Reply:
Addressed your other suggestions, thanks 🙏